### PR TITLE
Playwright: Setup video recording and screenshots in playwright

### LINKED ISSUE #14: Setup video recording and screenshots
## TODO

- [ ] Find a way to record videos of tests

- [ ] Store the screenshots and videos at `tests/selenium/log` folder. This should be readable from an environment variable

### PR DESCRIPTION
**Pull request by hariclerry**
### Tickets
Closes #14  

### What does this PR do?
- This PR setup video recording and screenshots

### What are the changes made?
- Installed playwright-video and FFmpeg plugins/dependencies
- Added video recording and screenshots to test suites

### Screen shots
None

### CODE REVIEW

File: tests/playwright/specs/wikipediaHomePage.js [hunk 1]
```diff
@@ -1,8 +1,28 @@
+const { saveVideo } = require( 'playwright-video' );
+const path = require( 'path' ),
+	LOGPATH = process.env.LOG_DIR || path.join( __dirname, '..', 'log' );
```
Comment (sovoid) [inline tests/playwright/specs/wikipediaHomePage.js]: use path.resolve with path.join. The generates the absolute path

File: tests/playwright/specs/wikipediaHomePage.js [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+
 describe( 'Page', () => {
 	beforeAll( async () => {
 		await page.goto( 'https://www.wikipedia.org/' );
 	} );
 
+	beforeEach( async () => {
+		capture = await saveVideo(
+			page,
+			`${LOGPATH}/recording.mp4`
```
Comment (sovoid) [inline tests/playwright/specs/wikipediaHomePage.js]: The order here should be

`beforeEach` -> test -> `afterEach`

this makes it more readable when reading the code sequentially.

Actually, just put them all in a single test

**Review by sovoid — CHANGES_REQUESTED** [review]

**Comment by hariclerry** [comment]
The feedback has been implemented @und3fined-v01d

File: tests/playwright/specs/wikipediaHomePage.js [hunk 3]
```diff
@@ -1,14 +1,22 @@
+const { saveVideo } = require( 'playwright-video' );
+const path = require( 'path' ),
+	LOGPATH = process.env.LOG_DIR || path.resolve( __dirname, '..', 'log' );
+
+let capture;
```
Comment (sovoid) [inline tests/playwright/specs/wikipediaHomePage.js]: lets move this declaration inside the `it` block as well

**Review by sovoid — CHANGES_REQUESTED** [review]

File: tests/playwright/specs/wikipediaHomePage.js (hunk 3 above)
Comment (hariclerry) [reply tests/playwright/specs/wikipediaHomePage.js]: This has been changed.

File: tests/playwright/specs/wikipediaHomePage.js [hunk 4]
```diff
@@ -1,14 +1,21 @@
+const { saveVideo } = require( 'playwright-video' );
+const path = require( 'path' ),
+	LOGPATH = process.env.LOG_DIR || path.resolve( __dirname, '..', 'log' );
```
Comment (zeljkofilipin) [inline tests/playwright/specs/wikipediaHomePage.js]: LOGPATH doesn't need to be all caps.

File: tests/playwright/specs/wikipediaHomePage.js [hunk 5]
```diff
@@ -1,14 +1,21 @@
...
-	beforeAll( async () => {
+	it( 'should display sub title correctly', async () => {
+		let capture;
 		await page.goto( 'https://www.wikipedia.org/' );
-	} );
 
-	afterAll( async () => {
-		await browser.close();
-	} );
+		capture = await saveVideo( page, `${LOGPATH}/recording.mp4` );
```
Comment (zeljkofilipin) [inline tests/playwright/specs/wikipediaHomePage.js]: Move this line to line 7, so video recording starts before the page is loaded.

**Review by zeljkofilipin — COMMENTED** [review]
Looks good in general! I've left a couple of inline comments.

File: tests/playwright/specs/wikipediaHomePage.js [hunk 6]
```diff
@@ -1,14 +1,21 @@
+const { saveVideo } = require( 'playwright-video' );
+const path = require( 'path' ),
+	LOGPATH = process.env.LOG_DIR || path.resolve( __dirname, '..', 'log' );
+
 describe( 'Page', () => {
-	beforeAll( async () => {
+	it( 'should display sub title correctly', async () => {
+		let capture;
```
Comment (zeljkofilipin) [inline tests/playwright/specs/wikipediaHomePage.js]: capture is maybe too generic name. captureVideo might be better.

**Review by zeljkofilipin — COMMENTED** [review]
I forgot one more thing. 😄

File: tests/playwright/specs/wikipediaHomePage.js (hunk 6 above)
Comment (hariclerry) [reply tests/playwright/specs/wikipediaHomePage.js]: This has been changed.

File: tests/playwright/specs/wikipediaHomePage.js (hunk 5 above)
Comment (hariclerry) [reply tests/playwright/specs/wikipediaHomePage.js]: This has been changed.

File: tests/playwright/specs/wikipediaHomePage.js (hunk 4 above)
Comment (hariclerry) [reply tests/playwright/specs/wikipediaHomePage.js]: This has been changed.

**Review by vidhi-mody — APPROVED** [review]

**Pull request merged**